### DISCUSSION TITLE
How to SpawnCommand with full environment (with PATH)?

### DISCUSSION THREAD

**Discussion by georgeguimaraes** [wezterm/wezterm#5203] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hi, I have this setup:

```lua
{
    mods = "CMD",
    key = ",",
    label = "Open Wezterm config",
    action = action.SpawnCommandInNewWindow({
      cwd = os.getenv("WEZTERM_CONFIG_DIR"),
      set_environment_variables = {
        PATH = os.getenv("PATH"),
      },
      args = {
        "/opt/homebrew/bin/nvim",
        os.getenv("WEZTERM_CONFIG_FILE"),
      },
    }),
  },
```

However, when I use it to open my wezterm config with nvim, nvim can't find some executables because the PATH is not set:

![SCR-20240322-ihrb](https://github.com/wez/wezterm/assets/2929/3479f241-6b1f-4de8-b049-b38eca84a98a)

Is there a way to have a full-blown zsh session there?

**Comment by wez** [comment, marked as answer]
See https://wezfurlong.org/wezterm/faq.html#im-on-macos-and-wezterm-cannot-find-things-in-my-path for an example of spawning using the shell explicitly, as well as background on what is happening here on macOS

**Reply by georgeguimaraes** [reply]
Tks, I tried several options, including using "zsh -ic" to force an interactive sessions, but the PATH came pretty basic as before.

So, I ended up using the last option, filling in the PATH manually. Not pretty, but it works :)

**Comment by daturkel** [comment]
The info in that FAQ may be out of date. I was not able to get it working with anything other than manually adding:

```
config.set_environment_variables = {
  PATH = '/opt/homebrew/bin:' .. os.getenv 'PATH',
}
```

then

```
  {
    key = ",",
    mods = "SUPER",
    action = wezterm.action.SpawnCommandInNewTab({
        args = { 'nvim', wezterm.config_file },
    }),
  },
```

The other options just led to a little screen flash whenever I hit command-comma.

This is on macos 14.7, `wezterm 20240203-110809-5046fc22`.

**Reply by syntruth** [reply]
Thank you! I did not relish using hardcoded paths, since my Wezterm goes across devices and MacOS/Linux.
